libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFDefaultUserAccountPersister.m

```diff
@@ -0,0 +1,306 @@
+                    NSString *orgPath = [rootPath stringByAppendingPathComponent:orgContent];
+
+                    // Now let's try to load the user account file in there
+                    NSString *userAccountPath = [orgPath stringByAppendingPathComponent:kUserAccountPlistFileName];
+                    if ([fm fileExistsAtPath:userAccountPath]) {
+                        SFUserAccount *userAccount = nil;
+                        [self loadUserAccountFromFile:userAccountPath account:&userAccount error:nil];
+                        if (userAccount) {
+                            userAccountMap[userAccount.accountIdentity] = userAccount;
+                        } else {
+                            // Error logging will already have occurred.  Make sure account file data is removed.
+                            [fm removeItemAtPath:userAccountPath error:nil];
+                        }
+                    } else {
+                        [self log:SFLogLevelDebug format:@"There is no user account file in this user directory: %@", orgPath];
+                    }
+                }
+            }
+        }
+    }
+    return userAccountMap;
+}
+
+- (BOOL)deleteAccountForUser:(SFUserAccount *)user error:(NSError **)error {
+    BOOL success = NO;
+    if (nil != user) {
+        NSFileManager *manager = [[NSFileManager alloc] init];
+        NSString *userDirectory = [[SFDirectoryManager sharedManager] directoryForUser:user
+                                                                                 scope:SFUserAccountScopeUser
+                                                                                  type:NSLibraryDirectory
+                                                                            components:nil];
+        if ([manager fileExistsAtPath:userDirectory]) {
+            NSError *folderRemovalError = nil;
+            success= [manager removeItemAtPath:userDirectory error:&folderRemovalError];
+            if (!success) {
+                [self log:SFLogLevelDebug
+                   format:@"Error removing the user folder for '%@': %@", user.userName, [folderRemovalError localizedDescription]];
+                if (folderRemovalError && error) {
+                    *error = folderRemovalError;
+                }
+            }
+        } else {
+            NSString *reason = [NSString stringWithFormat:@"User folder for user '%@' does not exist on the filesystem", user.userName];
+            NSError *ferror = [NSError errorWithDomain:SFUserAccountManagerErrorDomain
+                                                  code:SFUserAccountManagerCannotReadDecryptedArchive
+                                              userInfo:@{NSLocalizedDescriptionKey: reason}];
+            [self log:SFLogLevelDebug format:@"User folder for user '%@' does not exist on the filesystem.", user.userName];
+            if(error)
+                *error = ferror;
+            success = NO;
```

Comment:
Minor: don't really need this line.  Nothing would set `success` to `YES` at this point.